common/logging_extra.py

```diff
@@ -163,6 +163,16 @@ def event(self, event_name, *args, **kwargs):
     else:
       self.info(evt)
 
+  def blocking_info(self, msg, fsync_dir):
+    # if logmessaged is running
+    self.info(msg)
+    # else get_file_handler().emit(msg)
+    #   pass
+    fd = os.open(fsync_dir, os.O_RDONLY)
```

Comment:
You're probably looking for something like 
```
    for h in self.handlers:
      h.flush()
```

However, that also doesn't guarantee the message has been picked up by logmessaged already and has hit disk.

Reply:
So now I'm adding the file handler which bypasses logmessaged since it uses the filestream, calling handle which calls handle on all the handlers, which emits in all the handlers, but emit for the file_handler is just writing to a filestream and calling flush... then I make sure to fsync the whole directory